accumulo/src/main/java/com/yahoo/ycsb/db/accumulo/AccumuloClient.java

```diff
@@ -100,62 +104,72 @@ public void init() throws DBException {
   @Override
   public void cleanup() throws DBException {
     try {
-      if (bw != null) {
-        bw.close();
+      Iterator<BatchWriter> iterator = writers.values().iterator();
+      while (iterator.hasNext()) {
+        BatchWriter writer = iterator.next();
+        writer.close();
+        iterator.remove();
       }
     } catch (MutationsRejectedException e) {
       throw new DBException(e);
     }
   }
 
   /**
-   * Commonly repeated functionality: Before doing any operation, make sure
-   * we're working on the correct table. If not, open the correct one.
-   * 
-   * @param t
-   *          The table to open.
-   */
-  public void checkTable(String t) throws TableNotFoundException {
-    if (!table.equals(t)) {
-      getTable(t);
-    }
-  }
-
-  /**
    * Called when the user specifies a table that isn't the same as the existing
    * table. Connect to it and if necessary, close our current connection.
    * 
    * @param t
    *          The table to open.
    */
-  public void getTable(String t) throws TableNotFoundException {
-    if (bw != null) { // Close the existing writer if necessary.
-      try {
-        bw.close();
-      } catch (MutationsRejectedException e) {
-        // Couldn't spit out the mutations we wanted.
-        // Ignore this for now.
-        System.err.println("MutationsRejectedException: " + e.getMessage());
+  public BatchWriter getWriter(String t) throws TableNotFoundException {
+    // tl;dr We're paying a cost for the ConcurrentHashMap here to deal with the DB api.
+    // We know that YCSB is really only ever going to send us data for one table, so using
+    // a concurrent data structure is overkill (especially in such a hot code path).
+    // However, the impact seems to be relatively negligible in trivial local tests and it's
+    // "more correct" WRT to the API.
+    BatchWriter writer = writers.get(t);
```

Comment:
Did you look into using [MultiTableBatchWriter.getBatchWriter(String)][1]?  It does what you are doing here, but its performance may be worse.  

I looked at the MTBW implementation, it may do more reads to main memory that what you have in this patch.  The MTBW is trying to handle the case of a table being renamed, which causes more reads to main memory.

[1]: http://accumulo.apache.org/1.8/apidocs/org/apache/accumulo/core/client/MultiTableBatchWriter.html#getBatchWriter(java.lang.String)